src/rpdk/core/contract/resource_client.py

```diff
@@ -41,6 +41,50 @@ def prune_properties(document, paths):
     return document
 
 
+def prune_null_properties_in_path(model, paths):
+    """Prune given properties from a document.
+
+    This assumes properties will always have an object (dict) as a parent.
+    The function modifies the document in-place, but also returns the document
+    for convenience. (The return value may be ignored.)
+    """
+    document = {"properties": model.copy()}
+    for path in paths:
+        try:
+            prop, resolved_path, parent = traverse(document, path)
+        except LookupError:
+            pass  # not found means nothing to delete
+        else:
+            if not prop:
+                key = resolved_path[-1]
+                del parent[key]
+    return document["properties"]
+
+
+def prune_properties_from_model_in_path(output_model, input_model, paths):
+    """Prune given properties from a document.
+
+    This assumes properties will always have an object (dict) as a parent.
+    The function modifies the document in-place, but also returns the document
+    for convenience. (The return value may be ignored.)
+    """
+    output_document = {"properties": output_model.copy()}
```

Comment:
If this method is in-place modify, why we are doing a `.copy()`?

Reply:
I am trying to move away from in place modify to avoid unintended changes. Also to make sure the methos is self contained.